packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/components/Choices/Choices.js

```diff
@@ -93,41 +85,52 @@ export function Choices({
 
   return (
     <StyledGrid
+      role='menu'
       rowsCount={rowsCount}
     >
       {filteredChoiceIds.map((choiceId, index) => {
         const choice = task.choices?.[choiceId] || {}
         const selected = selectedChoiceIds.indexOf(choiceId) > -1
         const src = task.images?.[choice.images?.[0]] || ''
-        const hasFocus = autoFocus && (index === focusIndex)
-        const tabIndex = (index === focusIndex) ? 0 : -1
+        const hasFocus = !filterDropOpen && (index === focusIndex)
+        let tabIndex = -1
+        if (focusIndex === -1 && index === 0) {
+          tabIndex = 0
+        } else if (focusIndex === index) {
+          tabIndex = 0
+        }
 
         return (
-          <ChoiceButton
+          <li
             key={choiceId}
-            choiceId={choiceId}
-            choiceLabel={task.strings.get(`choices.${choiceId}.label`)}
-            disabled={disabled}
-            hasFocus={hasFocus}
-            onChoose={onChoose}
-            onKeyDown={handleKeyDown}
-            selected={selected}
-            src={src}
-            tabIndex={tabIndex}
-            thumbnailSize={thumbnailSize}
-          />
+            role="presentation"
+          >
+            <ChoiceButton
+              choiceId={choiceId}
```

Comment:
In terms of organising the code, I would declare the menu item roles and states here, as optional props for `ChoiceButton`, so that all the menu states are declared in the same component. It’s not a big deal, but choice button can’t be rendered in isolation otherwise eg. if you have a `ChoiceButton` story at the moment, you’re rendering a menu item in the browser without a wrapping menu. 

Of course, you could also update `ChoiceButton` stories and tests to make sure that the button is always wrapped in a menu. 

It’s the equivalent of writing a list item outside of a list, which is invalid HTML. 

```html
<li>this is invalid HTML</li>
```

If you aren’t rendering isolated menu items in stories and tests, you can ignore this comment.

Reply:
Yes that makes sense!